src/commons/Button.jsx

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+
+function Button({ onClick, children }) {
+  return (
+    <button
+      type="button"
+      onClick={() => onClick(children)}
```

Comment:
앞에서 고민하신 문제의 원인은 이곳 일 수 있을 것 같아요.

버튼 컴포넌트에서는 onClick으로 어떤 값이 넘어오던, 그냥 클릭이 되면 onClick 함수를 실행해주는 겁니다.
```suggestion
      onClick={onClick}
```

그리고 사용하는 쪽에서 어떤 클릭이 있을지를 넘겨주는 방식인거죠.
```javascript
// Counter.jsx -> 여기선 클릭시 1씩 더해줄 것이기 때문에 인자로 1을 넘겨줍니다.
<Button onClick={() => onClick(1)}>
```

Button 컴포넌트에서 children을 넘기다보니 버튼 컴포넌트를 사용하는 쪽에서 어떤식으로 사용할지 모르기 때문에, 우리의 의도와 다른 영향, 사이드 이펙트가 발생할 수 있을 것 같아요.

Reply:
부수효과를 줄이기 위해 onClick(children)이 아닌 count를 받아 onClick(count)가 될 수 있도록 변경 진행해봤습니다. 🙌